### PR TITLE
DEV-1460

### PR DESCRIPTION
**Pull request by ARatPeriship**
Created conditional statement to pull only minimum and maximum temperatures using Herbie.
Next would like to obtain specific coordinates from GRIB2 or json files to match with respective zip codes (if possible).
So far it works best when it pulls the latest model runs (-1 offset)

### CODE REVIEW

File: pytemp-extract/download_set.py [hunk 1]
```diff
@@ -8,9 +9,19 @@ def main(dt: datetime = typer.Option(datetime.utcnow(), help="Day to download"),
          offset: int = typer.Option(0, help="Time offset from dt in hours"),
          save_dir: str = typer.Option(".", help="Directory to save json output"),
          overwrite: bool = typer.Option(False, help="Overwrite existing downloaded files"),
-         download_only: bool = typer.Option(False, help="Download and do not process file")):
-
-    for fxx in range(1, 265):
+         download_only: bool = typer.Option(False, help="Download and do not process file"),
+         days: int = 5):
+
+    if datetime.utcnow().hour == 0:
```
[the remark this answers is not in the document]
Comment (PeterHu-Periship) [reply pytemp-extract/download_set.py:15]: Also you can set a variable like `current_utc_hour: int = datetime.now(timezone.utc)` before the if statements so you don't repeatedly call the same function.

File: pytemp-extract/download_set.py [hunk 2]
```diff
@@ -8,9 +9,19 @@ def main(dt: datetime = typer.Option(datetime.utcnow(), help="Day to download"),
...
+         init_hr: int = 12 
+    if datetime.utcnow().hour == 1:
+         init_hr: int = 12 - datetime.utcnow().hour 
+    if 2 <= datetime.utcnow().hour <= 13:
+         init_hr: int = 24 - datetime.utcnow().hour
+    else:
+        init_hr: int = 36 - datetime.utcnow().hour
```
[the remark this answers is not in the document]
Comment (javaknight) [reply pytemp-extract/download_set.py:22]: or use a ternary operator which is not included in the scope of our rule:

https://python-bloggers.com/2022/01/python-if-else-statement-in-one-line-ternary-operator-explained/

File: pytemp-extract/download_set.py [hunk 3]
```diff
@@ -8,9 +9,19 @@ def main(dt: datetime = typer.Option(datetime.utcnow(), help="Day to download"),
...
+
+    for fxx in range(init_hr - offset, 24 * days + init_hr - offset + 1,12): # Max is 265
```
[the remark this answers is not in the document]
Comment (dylanmanville) [reply pytemp-extract/download_set.py:24]: Yes, when setting variables, make them descriptive or add a line comment to describe what is happening

File: pytemp-extract/download_set.py [hunk 4]
```diff
@@ -1,18 +1,29 @@
...
 import process
 
 
-def main(dt: datetime = typer.Option(datetime.utcnow(), help="Day to download"),
+def main(dt: datetime = typer.Option(datetime.now(timezone.utc), help="Day to download"),
          offset: int = typer.Option(0, help="Time offset from dt in hours"),
          save_dir: str = typer.Option(".", help="Directory to save json output"),
          overwrite: bool = typer.Option(False, help="Overwrite existing downloaded files"),
-         download_only: bool = typer.Option(False, help="Download and do not process file")):
+         download_only: bool = typer.Option(False, help="Download and do not process file")) -> datetime:
```
Comment (PeterHu-Periship) [inline pytemp-extract/download_set.py:12]: Hm is datetime really the return type of this main() function?

File: pytemp-extract - Copy/process.py [hunk 5]
```diff
@@ -0,0 +1,119 @@
+import hashlib
```
Comment (carlos-sandoval-periship) [inline pytemp-extract - Copy/process.py:1]: Remove this file.   No need to create a copy.

[the remark this answers is not in the document]
Comment (ARatPeriship) [reply pytemp-extract - Copy/process.py]: That was a mistake on my local machine. Ignore the copy directory.

File: pytemp-extract/download_set.py [hunk 6]
```diff
@@ -1,18 +1,29 @@
...
+    init_hr: int = 36 - current_utc_hour
+
+    if current_utc_hour == 0:
+         init_hr: int = 12 
+    if current_utc_hour == 1:
+         init_hr: int = 12 - current_utc_hour 
+    if 2 <= current_utc_hour <= 13:
+         init_hr: int = 24 - current_utc_hour
+
+    for fcst_hr in range(init_hr - offset, 265,12): # Max is 265
```
[the remark this answers is not in the document]
Comment (ARatPeriship) [reply pytemp-extract/download_set.py:24]: Correct. I will update the name so it's not an abbreviation.

**Comment by PeterHu-Periship** [comment]
@ARatPeriship Also would like to suggest - when writing commit messages, try to make them more descriptive of what the main changes/goal of the commit is. For instance "Remove accidentally committed files" Or "Remove redundant 'copy' directory'". If all commit messages said something like "Update repo" it would be hard to follow the history of the changes

**Review by dylanmanville — APPROVED** [review]
